nan.h

```diff
@@ -1654,12 +1661,12 @@ class Callback {
       Execute(progress);
   }
 
-  void SendProgress_(const char *data, size_t size) {
-    char *new_data = new char[size];
+  void SendProgress_(T *data, size_t size) {
+    T *new_data = new T[size];
     memcpy(new_data, data, size);
```

Comment:
3rd argument of memcpy takes number of bytes but size is the number of elements now therefore this would work only for types with sizeof(T) == 1.